connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java

```diff
@@ -1252,10 +1200,14 @@ public void sendUpstreamPacketImmediately(BedrockPacket packet) {
      * @param packet the java edition packet from MCProtocolLib
      */
     public void sendDownstreamPacket(Packet packet) {
-        if (downstream != null && (protocol.getSubProtocol().equals(SubProtocol.GAME) || packet.getClass() == LoginPluginResponsePacket.class)) {
-            downstream.send(packet);
-        } else {
-            connector.getLogger().debug("Tried to send downstream packet " + packet.getClass().getSimpleName() + " before connected to the server");
+        if (!closed && this.downstream != null) {
+            this.downstream.getChannel().eventLoop().execute(() -> {
+                if (protocol.getSubProtocol().equals(SubProtocol.GAME) || packet.getClass() == LoginPluginResponsePacket.class) {
+                    downstream.send(packet);
+                } else {
+                    connector.getLogger().debug("Tried to send downstream packet " + packet.getClass().getSimpleName() + " before connected to the server");
+                }
+            });
```

Comment:
Might be worth adding a check to ensure you aren't running `this.downstream.getChannel().eventLoop().execute(...)` if the packet is already in the event loop.